tests/performance/density-kwok.go

```diff
@@ -0,0 +1,305 @@
+	node := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+			Labels: map[string]string{
+				"beta.kubernetes.io/arch":       "amd64",
+				"beta.kubernetes.io/os":         "linux",
+				"kubernetes.io/arch":            "amd64",
+				"kubernetes.io/hostname":        nodeName,
+				"kubernetes.io/os":              "linux",
+				"kubernetes.io/role":            "agent",
+				"node-role.kubernetes.io/agent": "",
+				"kubevirt.io/schedulable":       "true",
+				"type":                          "kwok",
+			},
+			Annotations: map[string]string{
+				"node.alpha.kubernetes.io/ttl": "0",
+				"kwok.x-k8s.io/node":           "fake",
+			},
+		},
+		Spec: corev1.NodeSpec{
+			Taints: []corev1.Taint{
+				{
+					Key:    "kwok.x-k8s.io/node",
+					Value:  "fake",
+					Effect: "NoSchedule",
+				},
+				{
+					Key:    "CriticalAddonsOnly",
+					Effect: corev1.TaintEffectNoSchedule,
+				},
+			},
+		},
+
+		Status: corev1.NodeStatus{
+			Allocatable: corev1.ResourceList{
+				corev1.ResourceCPU:              resource.MustParse("32"),
+				corev1.ResourceMemory:           resource.MustParse("256Gi"),
+				corev1.ResourceEphemeralStorage: resource.MustParse("100Gi"),
+				corev1.ResourcePods:             resource.MustParse("110"),
+				"devices.kubevirt.io/kvm":       resource.MustParse("1k"),
+				"devices.kubevirt.io/tun":       resource.MustParse("1k"),
+				"devices.kubevirt.io/vhost-net": resource.MustParse("1k"),
+			},
+			Capacity: corev1.ResourceList{
+				corev1.ResourceCPU:              resource.MustParse("32"),
+				corev1.ResourceMemory:           resource.MustParse("256Gi"),
+				corev1.ResourceEphemeralStorage: resource.MustParse("100Gi"),
+				corev1.ResourcePods:             resource.MustParse("110"),
+				"devices.kubevirt.io/kvm":       resource.MustParse("1k"),
+				"devices.kubevirt.io/tun":       resource.MustParse("1k"),
+				"devices.kubevirt.io/vhost-net": resource.MustParse("1k"),
+			},
+		},
+	}
+
+	return node
+}
+
+func createFakeVMIBatchWithKWOK(kubevirtClient kubecli.KubevirtClient, vmCount int) {
+	for i := 1; i <= vmCount; i++ {
+		vmi := createFakeVMISpecWithResources()
+
+		_, err := kubevirtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		time.Sleep(100 * time.Millisecond)
+	}
+}
+
+func deleteAndVerifyFakeVMIBatch(kubevirtClient kubecli.KubevirtClient, vmCount int, timeout time.Duration) {
+	err := kubevirtClient.VirtualMachineInstance(util.NamespaceTestDefault).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
+	if err != nil {
+		log.Fatal("Failed to delete VMIs")
+	}
+
+	Eventually(func() int {
+		vmis, err := kubevirtClient.VirtualMachineInstance(util.NamespaceTestDefault).List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		return len(vmis.Items)
+	}, timeout, 10*time.Second).Should(Equal(0))
```

Comment:
You can return the items slice and use the `BeEmpty` matcher.

Reply:
Done, updated. PTAL